### PR TITLE
Update compiler versions for neoverse v2

### PR DESCRIPTION
**Pull request by paolotricerri**
closes #96

Update compilers for v2 architecture.

### CODE REVIEW

**Comment by paolotricerri** [comment]
@dslarm

File: cpu/microarchitectures.json [hunk 1]
```diff
@@ -3102,15 +3102,19 @@
...
+                  "flags" : "-mcpu=neoverse-v2"
+              },
               {
-                  "versions": "12.0:12.99",
+                  "versions": "12.0:12.3.99",
                   "flags" : "-march=armv9-a+i8mm+bf16 -mtune=cortex-a710"
               },
 	      {
-                  "versions": "13.0:",
+                  "versions": "12.4:",
```
Comment (alalazo) [inline cpu/microarchitectures.json:3117]: Why are you suggesting 12.4, instead of 12.3 in #96 ?

Comment (dslarm) [reply cpu/microarchitectures.json:3117]: That was an error from our side, and #96 came in whilst this patch was being worked on.

Anything 12.3 and onwards will support -march=neoverse_v2

Per the Arm GCC team - the first releases with (including backports into 11 and 12) were:

- gcc 13.1.0, 
- gcc 12.3.0, 
- gcc 11.4.0 

@paolotricerri  - can you amend the GCC 12 one in the patch as it's incorrect, thanks?

Comment (paolotricerri) [reply cpu/microarchitectures.json]: @dslarm  and @alalazo I should have fixed the gcc version supporting neoverse-v2 from 12.3. I have also changed, accordingly, the versions range not supporting it above, that is, till 12.2.99

**Comment by Jordan474** [comment]
FWIW, LGTM, this can replace my #96 . I checked neoverse-v2 support was added looking at [gcc source code](https://gcc.gnu.org/git/?p=gcc.git;a=summary) for 12.3 and 11.4. And  I tested gcc 12.3 on neoverse-v2.

**Review by dslarm — APPROVED** [review]
LGTM - this is consistent with the info from the Arm GCC maintainers.

**Comment by paolotricerri** [comment]
Hi @alalazo,

As @dslarm  and @Jordan474  have agreed that this patch could address the wrong compiler versions on V2, I was wondering if there is an opportunity to complete the review?

Thank you for your understanding,

P

**Review by alalazo — APPROVED** [review]
Thanks, apologies for the delay in review - busy period :wink: !

**Pull request merged**